src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Providers/Providers/IaasVmPsBackupProvider.cs

```diff
@@ -1188,7 +1188,7 @@ private bool IsComputeAzureVM(string virtualMachineId)
             if (isDiscoveryNeeded)
             {
                 Logger.Instance.WriteDebug(string.Format(Resources.VMNotDiscovered, azureVMName));
-                RefreshContainer(vaultName: vaultName, resourceGroupName: resourceGroupName);
+                AzureWorkloadProviderHelper.RefreshContainer(ServiceClientAdapter, vaultName, resourceGroupName);
```

Comment:
Make this method non-static. Take ServiceClientAdapter as an argument to the ctor of AzureWorkloadProviderHelper. Initialize AzureWorkloadProviderHelper in the Init methods of the Azure VM and Azure File provider classes.